File: BeatPlannerAndroid/src/beatplanner/ui/views/MeterSpinner.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;

namespace BeatPlannerAndroid
{
  public abstract class MeterSpinner : Spinner
  {

    public MeterSpinner(Context context) :
      base(context)
    {
      Initialize();
    }

    public MeterSpinner(Context context, int mode) :
      base(context, SpinnerMode.Dialog)
    {
      Initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs) :
      base(context, attrs)
    {
      Initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs, int defStyle) :
      base(context, attrs, defStyle)
    {
      Initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs, int defStyle, int mode) :
      base(context, attrs, defStyle, SpinnerMode.Dialog)
    {
      Initialize();
    }

    protected abstract int TextArrayId { get; }

    protected virtual void Initialize()
    {
      var adapter = ArrayAdapter.CreateFromResource(
                      Context,
                      TextArrayId,
                      Android.Resource.Layout.SimpleSpinnerItem
                    );

      adapter.SetDropDownViewResource(Android.Resource.Layout.SimpleSpinnerDropDownItem);
      this.Adapter = adapter;
    }

  }
}

File: app/todo/ui/MeterSpinner.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:12
//

package dk.aschoen.beatplanner.ui;


public abstract class MeterSpinner  extends Spinner 
{
    public MeterSpinner(Context context) throws Exception {
        super(context);
        initialize();
    }

    public MeterSpinner(Context context, int mode) throws Exception {
        super(context, SpinnerMode.Dialog);
        initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs) throws Exception {
        super(context, attrs);
        initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs, int defStyle) throws Exception {
        super(context, attrs, defStyle);
        initialize();
    }

    public MeterSpinner(Context context, IAttributeSet attrs, int defStyle, int mode) throws Exception {
        super(context, attrs, defStyle, SpinnerMode.Dialog);
        initialize();
    }

    protected abstract int getTextArrayId() throws Exception ;

    protected void initialize() throws Exception {
        /* [UNSUPPORTED] 'var' as type is unsupported "var" */ adapter = ArrayAdapter.CreateFromResource(Context, getTextArrayId(), Android.Resource.Layout.SimpleSpinnerItem);
        adapter.SetDropDownViewResource(Android.Resource.Layout.SimpleSpinnerDropDownItem);
        this.Adapter = adapter;
    }

}


